# Preprocessing

In this notebook, I'll be preprocessing all my data to get it ready to be put into a model.

In [32]:
import pandas as pd
pd.set_option('display.max_columns', 100)
import numpy as np


## Data Loading

Just checking to see if my data has been correctly imported.

In [2]:
df=pd.read_csv("csvs/my_dataset.csv")

In [3]:
df.sample(10)

player_id  last_season_x  most_recent_club_id country_of_citizenship  \
3990       283225           2022                  678                Denmark   
84305       88682           2023                  506                  Italy   
120089     264646           2023                39097                Ukraine   
4105       290774           2023                11127                Croatia   
73161      533662           2023                  350                England   
51189      399262           2023                 2553               Scotland   
43907      513031           2023                  237                  Spain   
89193      651388           2021                61825                Croatia   
114600      95810           2023                  762                England   
40822       93853           2014                   86                Germany   

              sub_position  position   foot  height_in_cm  market_value  \
3990            Right-Back  Defender  right         181.0        600000   
84305           Right-Back  Defender  right         183.0       1800000   
120089           Left-Back  Defender   left         169.0        100000   
4105           Centre-Back  Defender  right         197.0       1800000   
73161       Centre-Forward    Attack  right         175.0       8000000   
51189        Left Midfield  Midfield  right         183.0        400000   
43907     Central Midfield  Midfield  right         179.0       1500000   
89193          Centre-Back  Defender  right         195.0        300000   
114600          Right-Back  Defender  right         173.0      18000000   
40822   Attacking Midfield  Midfield   both         178.0        600000   

        highest_ever_market_value        date  age  contract_days_left  month  \
3990                     800000.0  2021-10-06   24                 -98     10   
84305                  14000000.0  2012-06-08   20                 602      6   
120089                   200000.0  2015-03-15   19                 237      3   
4105                    2500000.0  2020-04-08   27                 237      4   
73161                  12000000.0  2022-11-03   21                1332     11   
51189                    450000.0  2022-12-13   24                 572     12   
43907                   2500000.0  2022-06-03   25                 237      6   
89193                    400000.0  2021-12-20   22                 237     12   
114600                 35000000.0  2021-12-30   31                 602     12   
40822                    800000.0  2014-07-10   20                 237      7   

        year  goals  assists  yellow_cards  red_cards  minutes_played  \
3990    2021      0        0             3          0            1195   
84305   2012      0        3             1          0            1496   
120089  2015      0        0             1          0             200   
4105    2020      1        0             3          1            1272   
73161   2022      0        0             0          0              52   
51189   2022      0        0             1          0             404   
43907   2022      0        0             7          0            1805   
89193   2021      0        0             6          0            1916   
114600  2021      0        2             1          0            2485   
40822   2014      0        0             1          0             121   

        player_club_id league_id  net_transfer_record  national_team_players  
3990               678       DK1           11930000.0                      3  
84305                5       IT1          -45000000.0                     15  
120089            6414      UKR1                  0.0                      0  
4105              2698       RU1           -6810000.0                      6  
73161              405       GB1          -60450000.0                     14  
51189             2553       SC1                  0.0                      2  
43907              237       ES1           -2300000.

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 123594 entries, 0 to 123593
Data columns (total 24 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   player_id                  123594 non-null  int64  
 1   last_season_x              123594 non-null  int64  
 2   most_recent_club_id        123594 non-null  int64  
 3   country_of_citizenship     123594 non-null  object 
 4   sub_position               123594 non-null  object 
 5   position                   123594 non-null  object 
 6   foot                       123594 non-null  object 
 7   height_in_cm               123594 non-null  float64
 8   market_value               123594 non-null  int64  
 9   highest_ever_market_value  123594 non-null  float64
 10  date                       123594 non-null  object 
 11  age                        123594 non-null  int64  
 12  contract_days_left         123594 non-null  int64  
 13  month                      12

In [5]:
df.describe()

player_id  last_season_x  most_recent_club_id   height_in_cm  \
count  1.235940e+05  123594.000000        123594.000000  123594.000000   
mean   2.455919e+05    2021.424948          3908.536029     182.265814   
std    1.784114e+05       2.298380          9918.900485       6.695002   
min    2.150000e+02    2012.000000             3.000000     159.000000   
25%    1.022260e+05    2021.000000           338.000000     178.000000   
50%    2.037460e+05    2023.000000           932.000000     183.000000   
75%    3.446050e+05    2023.000000          2700.000000     187.000000   
max    1.166093e+06    2023.000000         83678.000000     207.000000   

       market_value  highest_ever_market_value            age  \
count  1.235940e+05               1.235940e+05  123594.000000   
mean   5.106035e+06               1.030578e+07      25.031830   
std    1.061903e+07               1.747416e+07       3.985102   
min    1.000000e+04               2.500000e+04      15.000000   
25%    5.000000e+05               1.000000e+06      22.000000   
50%    1.500000e+06               3.500000e+06      25.000000   
75%    4.500000e+06               1.200000e+07      28.000000   
max    2.000000e+08               2.000000e+08      44.000000   

       contract_days_left          month           year          goals  \
count       123594.000000  123594.000000  123594.000000  123594.000000   
mean           448.834749       6.428346    2018.691587       1.973793   
std            471.763162       3.587796       2.976855       3.695028   
min           -309.000000       1.000000    2012.000000       0.000000   
25%            237.000000       3.000000    2017.000000       0.000000   
50%            237.000000       6.000000    2019.000000       1.000000   
75%            602.000000      10.000000    2021.000000       2.000000   
max           3159.000000      12.000000    2023.000000      59.000000   

             assists   yellow_cards      red_cards  minutes_played  \
count  123594.000000  123594.000000  123594.000000   123594.000000   
mean        1.501165       2.737706       0.068798     1308.635727   
std         2.450115       2.855257       0.265803     1010.055198   
min         0.000000       0.000000       0.000000        1.000000   
25%         0.000000       0.000000       0.000000      439.000000   
50%         1.000000       2.000000       0.000000     1143.000000   
75%         2.000000       4.000000       0.000000     1990.000000   
max        33.000000      23.000000       3.000000     5070.000000   

       player_club_id  net_transfer_record  national_team_players  
count   123594.000000         1.235940e+05          123594.000000  
mean      3175.928945         2.138397e+05               6.551669  
std       7846.777161         3.819746e+07               5.306323  
min          3.000000        -1.951000e+08               0.000000  
25%        294.000000        -1.930000e+06               2.000000  
50%        873.000000         1.500000e+05               6.000000  
75%       2457.000000         7.600000e+06              10.000000  
max      83678.000000         1.020000e+08              22.000000

Everything looks good, Lets get going with the preprocessing.

First, to avoid data leakage we will do our train test split before any proprocessing.

## Splitting

In [6]:
#Split my X and y
y=df["market_value"]
X=df.drop(["market_value"],axis=1)

In [7]:
#import splitter
from sklearn.model_selection import train_test_split

#split between train and val_test
X_train, X_val_test, y_train, y_val_test = train_test_split(X, y, test_size=0.33, random_state=100)

In [8]:
#split between test and val
X_val, X_test, y_val, y_test = train_test_split(X_val_test, y_val_test, test_size=0.5, random_state=100)

In [9]:
#check shape
X_train.shape

(82807, 23)

In [10]:
#check shape
X_val.shape

(20393, 23)

In [11]:
#check shape
X_test.shape

(20394, 23)

my splitting seems to have been correctly done. All the rows add up to the original size of the dataframe.

## Feature Encoding

Since I've got the month and year collumn i dont need the date collumn so I'll drop that.

In [12]:
X_train=X_train.drop(["date"],axis=1)
X_test=X_test.drop(["date"],axis=1)
X_val=X_val.drop(["date"],axis=1)

I need to encode the following collumns to be processed:

- `country_of_citizenship`
- `sub_position`
- `date`
- `position` 
- `foot` 
- `league_id`
  

We'll first look at the cardinality of all our categorical collumns to decide what type of encoding we should do.

In [13]:
df.nunique()

player_id                    13370
last_season_x                   12
most_recent_club_id            424
country_of_citizenship         155
sub_position                    13
position                         4
foot                             3
height_in_cm                    48
market_value                   223
highest_ever_market_value      157
date                          2141
age                             30
contract_days_left              89
month                           12
year                            12
goals                           51
assists                         31
yellow_cards                    23
red_cards                        4
minutes_played                4079
player_club_id                 426
league_id                       14
net_transfer_record            272
national_team_players           23
dtype: int64

the `foot`,`position`,`subposition` and `domestic competition id` collumns all have a low enough cardinality so one hot encoding would definitely be suitable.

In [14]:
#for train
encodecols = ["foot", "position", "sub_position", "league_id"]
X_train = pd.get_dummies(X_train, columns=encodecols, prefix=encodecols)

#for val
X_val=pd.get_dummies(X_val,columns=encodecols,prefix=encodecols)

#for test
X_test=pd.get_dummies(X_test,columns=encodecols,prefix=encodecols)

In [15]:
X_train.sample(5)

player_id  last_season_x  most_recent_club_id country_of_citizenship  \
31393     164292           2020                  338                  Spain   
19340     324344           2023                    3                 Greece   
99983     148344           2022                 1053                Denmark   
64652     346289           2023                 1049                 Guinea   
71392     289272           2023                  681                  Spain   

       height_in_cm  highest_ever_market_value  age  contract_days_left  \
31393         182.0                  6000000.0   28                -129   
19340         178.0                  4500000.0   19                 237   
99983         184.0                  2000000.0   18                 237   
64652         192.0                 30000000.0   24                1332   
71392         176.0                  2000000.0   25                 602   

       month  year  goals  assists  yellow_cards  red_cards  minutes_played  \
31393      4  2020      2        0             1          0             343   
19340     12  2016      1        1             1          0             839   
99983      7  2012      2        1             1          0             522   
64652     10  2020      1        0             5          0            2403   
71392     10  2020      0        0             0          0             368   

       player_club_id  net_transfer_record  national_team_players  foot_both  \
31393             338            1200000.0                      7          0   
19340            3060                  0.0                      5          0   
99983            1053             350000.0                      0          0   
64652            1049           14000000.0                      7          0   
71392             681          -10600000.0                      9          0   

       foot_left  foot_right  position_Attack  position_Defender  \
31393          0           1                1                  0   
19340          0           1                1                  0   
99983          0           1                1                  0   
64652          0           1                0                  1   
71392          1           0                1                  0   

       position_Goalkeeper  position_Midfield  \
31393                    0                  0   
19340                    0                  0   
99983                    0                  0   
64652                    0                  0   
71392                    0                  0   

       sub_position_Attacking Midfield  sub_position_Central Midfield  \
31393                                0                              0   
19340                                0                              0   
99983                                0                              0   
64652                                0                              0   
71392                                0                              0   

       sub_position_Centre-Back  sub_position_Centre-Forward  \
31393                         0                            1   
19340                         0                            0   
99983                         0                            0   
64652                         1                            0   
71392                         0                            0   

       sub_position_Defensive Midfield  sub_position_Goalkeeper  \
31393                                0                        0   
19340                                0                        0   
99983                                0                        0   
64652                                0                        0   
71392                                0                        0   

       sub_position_Left Midfield  sub_position_Left Winger  \
31393                           0                         0   
19340                           0                         0   
99983

In [16]:
X_test.sample(5)

player_id  last_season_x  most_recent_club_id country_of_citizenship  \
107939     137576           2023                 2282                Belgium   
57700       82442           2023                    5                 France   
72226      358719           2020                 2420                  Ghana   
121598      55439           2023                41231                 Russia   
33383      235568           2023                 3325               Portugal   

        height_in_cm  highest_ever_market_value  age  contract_days_left  \
107939         195.0                 16000000.0   27                1332   
57700          192.0                 27000000.0   34                 237   
72226          177.0                   800000.0   24                -129   
121598         184.0                  1200000.0   25                 237   
33383          177.0                  1500000.0   25                 237   

        month  year  goals  assists  yellow_cards  red_cards  minutes_played  \
107939      6  2019     17        3             8          0            3663   
57700       3  2021      6        1             3          0            1475   
72226       6  2018      0        0             0          0             126   
121598      6  2015      0        0             0          1            1017   
33383       5  2018      1        0             0          0             561   

        player_club_id  net_transfer_record  national_team_players  foot_both  \
107939            2282           21850000.0                      9          0   
57700              631         -195100000.0                     19          0   
72226              294          -49420000.0                      9          0   
121598           11127            -770000.0                      4          0   
33383              441                  0.0                      0          0   

        foot_left  foot_right  position_Attack  position_Defender  \
107939          0           1                0                  0   
57700           1           0                1                  0   
72226           0           1                0                  0   
121598          0           1                0                  0   
33383           0           1                0                  0   

        position_Goalkeeper  position_Midfield  \
107939                    0                  1   
57700                     0                  0   
72226                     0                  1   
121598                    1                  0   
33383                     0                  1   

        sub_position_Attacking Midfield  sub_position_Central Midfield  \
107939                                1                              0   
57700                                 0                              0   
72226                                 0                              1   
121598                                0                              0   
33383                                 0                              1   

        sub_position_Centre-Back  sub_position_Centre-Forward  \
107939                         0                            0   
57700                          0                            1   
72226                          0                            0   
121598                         0                            0   
33383                          0                            0   

        sub_position_Defensive Midfield  sub_position_Goalkeeper  \
107939                                0                        0   
57700                                 0                        0   
72226                                 0                        0   
121598                                0                        1   
33383                                 0                        0   

        sub_position_Left Midfield  sub_position_Left Winger  \
107939                           0                         0   
57700        

In [17]:
X_val.sample(5)

player_id  last_season_x  most_recent_club_id country_of_citizenship  \
89928     171068           2021                 2441               Portugal   
34220     332889           2023                  244                 France   
49218     522832           2023                39097                Ukraine   
9164      129444           2023                12321                  Spain   
60081       5023           2020                  506                  Italy   

       height_in_cm  highest_ever_market_value  age  contract_days_left  \
89928         179.0                  2000000.0   26                -129   
34220         172.0                 22000000.0   24                 967   
49218         182.0                   300000.0   28                 602   
9164          185.0                 15000000.0   33                 237   
60081         192.0                 35000000.0   42                 237   

       month  year  goals  assists  yellow_cards  red_cards  minutes_played  \
89928      7  2014      0        2             8          0            2202   
34220      6  2018      2        1             3          0            2815   
49218      9  2023      0        0             6          0             841   
9164       6  2022      2        1             3          0             992   
60081     12  2019      0        0             0          0            1710   

       player_club_id  net_transfer_record  national_team_players  foot_both  \
89928            2995                  0.0                      1          0   
34220             995            -450000.0                      7          0   
49218           39097                  0.0                      0          0   
9164              714            5500000.0                      4          0   
60081             583         -146500000.0                     18          0   

       foot_left  foot_right  position_Attack  position_Defender  \
89928          1           0                0                  1   
34220          0           1                0                  0   
49218          0           1                0                  0   
9164           0           1                0                  1   
60081          0           1                0                  0   

       position_Goalkeeper  position_Midfield  \
89928                    0                  0   
34220                    0                  1   
49218                    0                  1   
9164                     0                  0   
60081                    1                  0   

       sub_position_Attacking Midfield  sub_position_Central Midfield  \
89928                                0                              0   
34220                                0                              1   
49218                                0                              0   
9164                                 0                              0   
60081                                0                              0   

       sub_position_Centre-Back  sub_position_Centre-Forward  \
89928                         0                            0   
34220                         0                            0   
49218                         0                            0   
9164                          1                            0   
60081                         0                            0   

       sub_position_Defensive Midfield  sub_position_Goalkeeper  \
89928                                0                        0   
34220                                0                        0   
49218                                1                        0   
9164                                 0                        0   
60081                                0                        1   

       sub_position_Left Midfield  sub_position_Left Winger  \
89928                           0                         0   
34220                           0                         0   
49218

This seems to have worked well.The only collumn i need to deal with now is the `country_of_citizenship` collumn which consists of 155 unique values which on its own is far too many to one hot encode. The likely solution to this would be to get the top N-countries of citizenship in terms of numbers and have one designated collumn named other for every other nation other than them. This seems viable as nationalities in this dataset arent uniformly distributed as seen in our EDA in the previous notebook as the top 10 nationalities accounted for a disproportionate amount of the dataset.

In [18]:
#get the value counts for each nation
country_counts = df["country_of_citizenship"].value_counts()

#get the top 50 and its index
top_countries=country_counts.head(35).index



In [19]:
top_countries

Index(['Spain', 'France', 'Netherlands', 'Brazil', 'Turkey', 'Germany',
       'Portugal', 'Italy', 'Russia', 'England', 'Ukraine', 'Belgium',
       'Denmark', 'Greece', 'Argentina', 'Scotland', 'Serbia', 'Croatia',
       'Senegal', 'Nigeria', 'Sweden', 'Morocco', 'Uruguay', 'Cote d'Ivoire',
       'Poland', 'Ghana', 'Switzerland', 'Norway', 'Colombia', 'Austria',
       'Bosnia-Herzegovina', 'Cameroon', 'Algeria', 'Japan', 'United States'],
      dtype='object')

I need to now set everything not in that index to "Other".

In [20]:
#create lambda function setting "Other" if not in index for train,test and val
X_train["country_of_citizenship"]=X_train["country_of_citizenship"].apply(lambda x: x if x in top_countries else 'Other')
X_val["country_of_citizenship"]=X_val["country_of_citizenship"].apply(lambda x: x if x in top_countries else 'Other')
X_test["country_of_citizenship"]=X_test["country_of_citizenship"].apply(lambda x: x if x in top_countries else 'Other')

In [21]:
X_train.head()

player_id  last_season_x  most_recent_club_id country_of_citizenship  \
76575      111227           2022                 1038                  Italy   
115759     162074           2021                  714                  Spain   
38683      149446           2022                  605                  Other   
44111      568158           2023                 1108                  Spain   
90042      196435           2022                   80                  Other   

        height_in_cm  highest_ever_market_value  age  contract_days_left  \
76575          184.0                 18000000.0   24                 602   
115759         180.0                  5000000.0   20                -129   
38683          187.0                  4500000.0   26                 237   
44111          183.0                   800000.0   18                 602   
90042          181.0                  3000000.0   22                1332   

        month  year  goals  assists  yellow_cards  red_cards  minutes_played  \
76575       6  2018      0        0             0          0              23   
115759      1  2015      1        0             0          0             161   
38683       6  2018      2        0             6          0             810   
44111       2  2020      0        0             1          0             281   
90042      10  2017      0        0             4          0            1211   

        player_club_id  net_transfer_record  national_team_players  foot_both  \
76575                5          -45000000.0                     15          0   
115759             150           43500000.0                      7          0   
38683              618           12850000.0                      3          0   
44111             1108           -6300000.0                      3          0   
90042              618           12850000.0                      3          0   

        foot_left  foot_right  position_Attack  position_Defender  \
76575           0           1                0                  1   
115759          0           1                1                  0   
38683           0           1                0                  0   
44111           1           0                0                  1   
90042           0           1                0                  1   

        position_Goalkeeper  position_Midfield  \
76575                     0                  0   
115759                    0                  0   
38683                     0                  1   
44111                     0                  0   
90042                     0                  0   

        sub_position_Attacking Midfield  sub_position_Central Midfield  \
76575                                 0                              0   
115759                                0                              0   
38683                                 0                              1   
44111                                 0                              0   
90042                                 0                              0   

        sub_position_Centre-Back  sub_position_Centre-Forward  \
76575                          0                            0   
115759                         0                            0   
38683                          0                            0   
44111                          0                            0   
90042                          0                            0   

        sub_position_Defensive Midfield  sub_position_Goalkeeper  \
76575                                 0                        0   
115759                                0                        0   
38683                                 0                        0   
44111                                 0                        0   
90042                                 0                        0   

        sub_position_Left Midfield  sub_position_Left Winger  \
76575                            0                         0   
115759       

I can now one hot encoding without excessively increasing dimensionality.

In [22]:
#train
X_train = pd.get_dummies(X_train, columns=["country_of_citizenship"], prefix="Country")

#val
X_val = pd.get_dummies(X_val, columns=["country_of_citizenship"], prefix="Country")

#test
X_test = pd.get_dummies(X_test, columns=["country_of_citizenship"], prefix="Country")


In [23]:
X_train.sample(5)

player_id  last_season_x  most_recent_club_id  height_in_cm  \
5740      610442           2023                  985         191.0   
22951     122155           2023                  150         189.0   
4977      411295           2023                  131         176.0   
51433     442319           2021                49702         183.0   
88862     550549           2023                  232         182.0   

       highest_ever_market_value  age  contract_days_left  month  year  goals  \
5740                  45000000.0   18                1698     12  2020      0   
22951                 30000000.0   29                 967      3  2021      9   
4977                  60000000.0   22                1332      6  2019      8   
51433                   500000.0   24                -129      6  2022      1   
88862                  8000000.0   22                 967      6  2023      1   

       assists  yellow_cards  red_cards  minutes_played  player_club_id  \
5740         0             0          0             238             190   
22951        7             0          0            2390             681   
4977         3            10          0            2646             336   
51433        2             3          0             876           49702   
88862        1             0          0            1229             232   

       net_transfer_record  national_team_players  foot_both  foot_left  \
5740             9630000.0                      9          0          1   
22951          -10600000.0                      9          0          0   
4977            69500000.0                      8          0          1   
51433            1030000.0                      7          0          0   
88862          -19500000.0                     13          0          0   

       foot_right  position_Attack  position_Defender  position_Goalkeeper  \
5740            0                1                  0                    0   
22951           1                1                  0                    0   
4977            0                1                  0                    0   
51433           1                1                  0                    0   
88862           1                0                  1                    0   

       position_Midfield  sub_position_Attacking Midfield  \
5740                   0                                0   
22951                  0                                0   
4977                   0                                0   
51433                  0                                0   
88862                  0                                0   

       sub_position_Central Midfield  sub_position_Centre-Back  \
5740                               0                         0   
22951                              0                         0   
4977                               0                         0   
51433                              0                         0   
88862                              0                         0   

       sub_position_Centre-Forward  sub_position_Defensive Midfield  \
5740                             1                                0   
22951                            1                                0   
4977                             0                                0   
51433                            1                                0   
88862                            0                                0   

       sub_position_Goalkeeper  sub_position_Left Midfield  \
5740                         0                           0   
22951                        0                           0   
4977                         0                           0   
51433                        0                           0   
88862                        0                           0   

       sub_position_Left Winger  sub_position_Left-Back  \
5740                          0                       0   
22951                         0               

I've now completed my feature encoding and can move onto the next step of my preprocessing.

## Scaling Data

I'll use the StandardScaler as there will be quite a few outliers as bigger teams in better leagues have a disproportionate amount of funds to spend on players which min-max scaling is very sensitive to.

In [27]:
from sklearn.preprocessing import StandardScaler
#instantiate
scaler=StandardScaler()

#fit and transform
#train
X_train_s=scaler.fit_transform(X_train)

#val
X_val_s=scaler.fit_transform(X_val)


#test
X_test_s=scaler.fit_transform(X_test)




They're now in array form and I want it back in dataframe format.

In [29]:
#train
X_train_s = pd.DataFrame(X_train_s, columns=X_train.columns)

#val
X_val_s = pd.DataFrame(X_val_s, columns=X_val.columns)


#test
X_test_s = pd.DataFrame(X_test_s, columns=X_test.columns)



In [30]:
X_train_s

player_id  last_season_x  most_recent_club_id  height_in_cm  \
0      -0.750024       0.251602            -0.287729      0.260554   
1      -0.464942      -0.183373            -0.320548     -0.337849   
2      -0.535743       0.251602            -0.331589      0.709356   
3       1.811829       0.686577            -0.280639      0.110954   
4      -0.272292       0.251602            -0.384767     -0.188248   
...          ...            ...                  ...           ...   
82802  -0.397551       0.686577            -0.350632     -0.337849   
82803  -1.073477       0.251602            -0.351341     -0.337849   
82804   0.397310       0.686577             0.273026      0.410155   
82805   0.075544       0.686577            -0.280639      1.008558   
82806  -0.616832      -0.183373            -0.317003      1.158159   

       highest_ever_market_value       age  contract_days_left     month  \
0                       0.440320 -0.260559            0.325471 -0.118134   
1                      -0.304066 -1.262690           -1.223367 -1.511990   
2                      -0.332696  0.240506           -0.447889 -0.118134   
3                      -0.544560 -1.763756            0.325471 -1.233219   
4                      -0.418587 -0.761625            1.872191  0.996950   
...                          ...       ...                 ...       ...   
82802                   1.012924 -1.513223           -0.447889 -1.511990   
82803                  -0.160915  1.994235            0.325471 -0.118134   
82804                   0.554841 -0.511092            1.872191 -0.118134   
82805                  -0.475847  0.491039            1.872191  1.554492   
82806                  -0.447217  1.242637           -1.223367 -0.118134   

           year     goals   assists  yellow_cards  red_cards  minutes_played  \
0     -0.230274 -0.532255 -0.612944     -0.960436  -0.258501       -1.272918   
1     -1.237884 -0.262411 -0.612944     -0.960436  -0.258501       -1.136201   
2     -0.230274  0.007433 -0.612944      1.148781  -0.258501       -0.493237   
3      0.441465 -0.532255 -0.612944     -0.608900  -0.258501       -1.017317   
4     -0.566144 -0.532255 -0.612944      0.445709  -0.258501       -0.095967   
...         ...       ...       ...           ...        ...             ...   
82802 -1.573754  0.547121  0.201472     -0.257364  -0.258501        1.525808   
82803  1.449075 -0.532255 -0.612944     -0.257364  -0.258501       -0.609149   
82804  1.113205  1.896341 -0.205736      0.445709  -0.258501        1.417822   
82805  1.113205 -0.532255 -0.612944      0.094172  -0.258501       -0.454600   
82806  0.777335  0.007433 -0.205736     -0.960436   3.500180       -1.115396   

       player_club_id  net_transfer_record  national_team_players  foot_both  \
0           -0.402521            -1.179271               1.592686  -0.203311   
1           -0.383949             1.134991               0.082045  -0.203311   
2           -0.324004             0.333498              -0.673275  -0.203311   
3           -0.261241            -0.167272              -0.673275  -0.203311   
4           -0.324004             0.333498              -0.673275  -0.203311   
...               ...                  ...                    ...        ...   
82802       -0.279045             0.829038              -0.106785  -0.203311   
82803       -0.350646             0.221315               0.270875  -0.203311   
82804       -0.307224             1.090536               0.082045  -0.203311   
82805       -0.356154            -0.305866              -0.106785  -0.203311   
82806       -0.307224             1.090536               0.082045  -0.203311   

       foot_left  foot_right  position_Attack  position_Defender  \
0      -0.577466    0.638745        -0.635055           1.407452   
1      -0.577466    0.638745         1.574668          -0.710504   
2      -0.577466    0.638745        -0.635055          -0.710504   
3       1.731702   -1.565570        -0.635055           1.4

Data looks correctly scaled. We can now move onto our Baseline linear Model.

### Data Export

In [33]:
import pickle
X_train_s.to_pickle('X_train_s.pkl')
X_val_s.to_pickle('X_val_s.pkl')
X_test_s.to_pickle('X_test_s.pkl')
y_train.to_pickle('y_train.pkl')
y_val.to_pickle('y_val.pkl')
y_test.to_pickle('y_test.pkl')